### DISCUSSION TITLE
many to many relation

### DISCUSSION THREAD

**Discussion by DarkPhoenix2704** [prisma/prisma#16236] [Q&A] [answered]
I have actually have some issues with creating model for my database.
i have 3 tables, users, teams, events

A user can create an team for an event
A team should have a lead & 2 members
A user should not have more than 1 Team for an event.
```
model User {
  id              String   @id @unique @default(cuid()) @db.VarChar(30)
  name            String
  teams           Team[]
}

model Event {
  id            String      @id @unique @default(cuid()) @db.VarChar(30)
  name          String
  date          DateTime
  teams         Team[]
  projectCount  Int         @default(0)
}
model Team {
  id            String      @id @unique @default(cuid()) @db.VarChar(30)
  name          String
  eventid       String
  event         Event       @relation(fields: [eventid], references: [id])
  members       User[]      
}
```
Currently this is the model I am using.
But i am unable to add leader relation & one team per event per user

**Comment by pantharshit00** [comment, marked as answer]
Hello @DarkPhoenix2704 

I think you can use a relation table here to store this three way relation:
```prisma
model User {
  id             String            @id @unique @default(cuid()) @db.VarChar(30)
  name           String
  teamsAndEvents UserToTeamEvent[]
}

model UserToTeamEvent {
  id String @id @default(cuid()) @db.VarChar(30)

  isLeader Boolean
  team     Team    @relation(fields: [teamId], references: [id])
  user     User    @relation(fields: [userId], references: [id])
  event    Event   @relation(fields: [eventId], references: [id])

  eventId String @db.VarChar(30)
  teamId  String @db.VarChar(30)
  userId  String @db.VarChar(30)

  @@unique([userId, teamId, eventId])
}

model Event {
  id           String            @id @unique @default(cuid()) @db.VarChar(30)
  name         String
  date         DateTime
  userAndTeam  UserToTeamEvent[]
  projectCount Int               @default(0)
}

model Team {
  id           String            @id @unique @default(cuid()) @db.VarChar(30)
  name         String
  eventid      String
  userAndEvent UserToTeamEvent[]
}

```

Let me know if this will solve your use case otherwise we can tweak it further.

**Reply by DarkPhoenix2704** [reply]
But still a user can create more than a single team for an event?

**Reply by pantharshit00** [reply]
You can add another `@@unique([userId, eventId])` to prevent that. That way user can only join that event once.

**Reply by DarkPhoenix2704** [reply]
okey

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
